cpp/src/arrow/compute/key_map_internal.h

```diff
@@ -220,6 +257,12 @@ class ARROW_EXPORT SwissTable {
     return bits_stamp_;
   }
 
+  static uint32_t group_id_mask_from_num_groupid_bits(int64_t num_groupid_bits) {
+    return static_cast<uint32_t>((1ULL << num_groupid_bits) - 1);
+  }
+
+  static constexpr int bytes_status_in_block_ = 8;
```

Comment:
Usually, compile-time constants should follow the naming convention `kBytesStatusInBlock`. Perhaps we can have a renaming pass in this PR or another one?

Reply:
Yes it is supposed to be. However I was also following the naming convention of several existing compile time constants in this class. I would like to to change them all in another PR to keep this one solely focused on the purpose the overflow prevention.